src/view/com/modals/Modal.web.tsx

```diff
@@ -51,23 +51,36 @@ function Modal({modal}: {modal: ModalIface}) {
   const pal = usePalette('default')
   const {isMobile} = useWebMediaQueries()
 
+  const clickInsideRef = React.useRef<boolean>(false)
+
   if (!isModalActive) {
     return null
   }
 
-  const onPressMask = () => {
+  const onPressMask = (ev: React.MouseEvent<HTMLDivElement>) => {
+    // 1. Ignore clicks from inside the modal
+    // 2. Ignore clicks that started inside the modal and ended in the mask
+    if (ev.target !== ev.currentTarget.firstChild || clickInsideRef.current) {
```

Comment:
Any solution that relies on timing here (like with tracking pointer down/up) seems suspicious to me. For this type of problem it's usually sufficient to check whether one node `contains` the other node.

What is special that requires this?

Reply:
Yeah this doesn't seem the right approach I think, now that I've tested it again this actually swallows clicks to backdrop if you do the second case.

Reply:
Hm, it seems that Headless UI does more or less the same thing to check where the click initially started.

https://github.com/tailwindlabs/headlessui/blob/2d5d35a5337154e823767291637634b69b09ab94/packages/%40headlessui-react/src/hooks/use-outside-click.ts#L106-L149

The problem this PR tries to fix can be illustrated here:

https://github.com/bluesky-social/social-app/assets/148872143/e5b292eb-71ff-4d5d-8740-6496f061e89d

It's very easy to overshoot the mouse cursor when selecting text, and while a confirmation modal would've easily done the same, it doesn't really remove the annoyance issue that overshooting causes